### ISSUE TITLE
Getting Primary Accelerometer Data

### ISSUE THREAD

**Issue by shanlyMIDE** [MideTechnology/endaq-python-ide#47]
The cloud scripts will need to pull out the _primary_ accelerometer data, defined by the one with the highest number of samples. The struggle is though that sometimes this can be channel 8 which also contains the microphone data.

We have nice functions to filter subchannels by acceleration type, but then it isn't all that obvious how to use this to get the actual acceleration data. They could loop through and to `to_pandas()` on every subchannel but this would be needlessly recomputing timestamps.

Here's an option using the channel table to filter by units and by number of samples, this would be nice to wrap into a function.

~~~python
#Import file
doc = endaq.ide.get_doc(file_download_url)

#Get accel
table = endaq.ide.get_channel_table(doc) #get summary of all channels
accel_channels = table.data[table.data.units=='g'].copy() #filter to only acceleration
accel_channels = accel_channels[accel_channels.samples==accel_channels.samples.max()] #filter to highest number of samples
accel = endaq.ide.to_pandas(accel_channels.channel.iloc[0].parent) #get channel data
accel = accel[accel_channels.name] #filter away non accelerometer channels (if they exist)
~~~

Here's a good test file that has all channels on: `'https://info.endaq.com/hubfs/data/All-Channels.ide'`

**Comment by SamRagusa** [comment]
One way of handling this a bit better could be setting the `measurement_type` parameter of the `get_channel_table` function, specifically you'd want to set it to `endaq.ide.ACCELERATION`.  This should only get the channel data for acceleration based data.

(warning: I haven't actually ever used this parameter, just know that it's there)

**Comment by shanlyMIDE** [comment]
yes this will work, but then I will need to loop through every subchannel to get the data which is fine for us to handle under-the-hood but not make the user do

**Comment by CrepeGoat** [comment]
I *want* to say we discussed this before, back when David and I both wrote code to pull out channel data. I think I brought up this scenario, and we decided the solution was to get the accel channel, call `to_pandas` on the whole channel, and just remove the mic column from the `pd.DataFrame` (so like `del df["Mic"]` or something?)

**Comment by shanlyMIDE** [comment]
Yeah that's the approach I took although I did it a little different to not hardcode the `"Mic"` channel because:
- I've anecdotally noticed we haven't been consistent with our name there
- In the future (think H series) there will be other analog channels that aren't acceleration with unknown subchannel names

So instead I:
1. Found the accelerometer _subchannel_ with highest number of samples  
2. Got that subchannel's _parent_ channel data
3. Filtered the dataframe to only contain the subchannel names that have units of acceleration

**Comment by CrepeGoat** [comment]
btw this sounds like @StokesMIDE might have better insight into how to do this more cleanly / in fewer lines of code?

**Comment by StokesMIDE** [comment]
There are a couple of components to this (not in order):
* Get accelerometer subchannels
* Identify the 'primary' accelerometer
* Create a Dataframe that excludes non-acceleration subchannels (and/or creates a Dataframe from a list of subchannels)

Getting accelerometer subchannels is pretty easy:
```python
doc = endaq.ide.get_doc(file_download_url)
accel_channels = endaq.ide.get_channels(doc, endaq.ide.ACCELERATION)
```

I've got some ideas for nicely implementing how to find the primary accelerometer, but I need to think about it a little more.

In terms of creating the Dataframe, maybe `endaq.ide.to_pandas()` could be modified (or an alternate version implemented) that takes a list of subchannel IDs as an argument (like `idelib.dataset.Dataset.exportCsv()`).

Incidentally, there's an [`idelib` issue](https://github.com/MideTechnology/idelib/issues/77) that seems fairly similar.
